cache/sampleStoreInit.js

```diff
@@ -160,63 +193,76 @@ function populateSamples() {
-  if (!featureToggles.isFeatureEnabled(constants.featureName)) {
-    return Promise.resolve(false);
-  }
-
-  const msg = 'Populating redis sample store from db';
-  winston.info(msg);
+  const msg = 'Populating redis sample store from db started :|';
+  log.info(msg);
 
   const promises = [populateSubjects(), populateSamples(), populateAspects()];
   return Promise.all(promises);
 } // populate
 
 /**
- * Checks whether the sample store exists by counting the members of the three
- * index keys (samples, subjects, aspects). If any has more than one member,
- * consider the sample store as existing.
  *
- * Note: I tried using the redis "exists" command but our travis build kept
- * failing with a message that I was passing the wrong number of args to the
- * "exists" command. I gave up and went with counting members using "scard"
- * instead.
+ * Compare the current status of the "enableRedisSampleStore" flag with its
+ * previous status. If both current status and previous status are the same, do
+ * nothing. When the current status and the previous status do not match, take
+ * some action depending on the current status.
  *
  * @returns {Promise} which resolves to true if the sample store already
  *  exists.
  */
-function sampleStoreExists() {
-  const cmds = [
-    ['scard', constants.indexKey.aspect],
-    ['scard', constants.indexKey.sample],
-    ['scard', constants.indexKey.subject],
-  ];
-  return redisClient.batch(cmds).execAsync()
-  .then((batchResponse) =>
-    batchResponse[0] > 0 || // eslint-disable-line no-magic-numbers
-    batchResponse[1] > 0 || // eslint-disable-line no-magic-numbers
-    batchResponse[2] > 0); // eslint-disable-line no-magic-numbers
-} // sampleStoreExists
+function storeSampleToCacheOrDb() {
+  const currentStatus = featureToggles.isFeatureEnabled(constants.featureName)
+                                          || false;
+  return getPreviousStatus()
+  .then((prevState) => {
+    const previousStatus = (prevState == 'true') || false;
+
+    // set the previousStatus to the currentStatus
+    redisClient.setAsync(constants.previousStatusKey, currentStatus);
+
+    /*
+     * when the current status and the previous status do not match, actions
+     * needs to be taken based on the current status
+    */
+    if (previousStatus !== currentStatus) {
+      /*
+       * call "popluate" when "enableRedisSampleStore" flag has been changed
+       * from false to true. Call "eradicate" and "storeSampleToDb" when
+       * "enableRedisSampleStore" flag has been changed from true to false
+       */
+      if (currentStatus) {
+        log.info('"enableRedisSampleStore" flag was switched to true,' +
+          ' so populating the cache from db');
+        return populate();
+      }
+
+      log.info('"enableRedisSampleStore" flag was switched to false' +
+          ' so persisting to db from cache. The cache will be eradicated ' +
+          'after the samples are persisted to db');
```

Comment:
I have added quite a bit of logging. These get logged when running the tests too, so if we think that it is too much, I can remove some.